src/tools/tidy/src/ui_tests.rs

```diff
@@ -73,6 +79,56 @@ pub fn check(root_path: &Path, tidy_ctx: TidyCtx) {
             ));
         }
     }
+
+    // The list of subdirectories in ui tests.
+    // Compare previous subdirectory with current subdirectory
+    // to sync with `tests/ui/README.md`.
+    // See <https://github.com/rust-lang/rust/issues/150399>
+    let mut prev_line = "";
+    let mut is_sorted = true;
+    let prev_subdirs: BTreeSet<_> = include_str!("ui_suites.txt")
+        .strip_prefix(UI_SUITES_TXT_HEADER)
+        .unwrap()
+        .lines()
+        .inspect(|&line| {
+            if prev_line > line {
+                is_sorted = false;
+            }
+            prev_line = line;
+        })
+        .map(|line| line.to_string())
+        .collect();
+
+    let is_modified = !cur_subdirs.eq(&prev_subdirs);
+
+    if !is_sorted && !bless {
+        check.error(
+            "`src/tools/tidy/src/ui_suites.txt` is not in order, mostly because you modified it manually,
+            please only update it with command `x test tidy --bless`"
+        );
+    } else if is_modified && !bless {
+        for directory in prev_subdirs.symmetric_difference(&cur_subdirs) {
+            if prev_subdirs.contains(directory) {
+                check.error(format!("ui subdirectory `tests/{directory}` deleted"));
+            } else {
+                check.error(format!("ui subdirectory `tests/{directory}` created"));
+            }
+        }
+        check.error("please update `tests/ui/README.md` and use `x test tidy --bless` command");
+    } else if bless && (!is_sorted || is_modified) {
+        let tidy_src = root_path.join("src/tools/tidy/src");
+        // instead of overwriting the file, recreate it and use an "atomic rename"
+        // so we don't bork things on panic or a contributor using Ctrl+C
+        let blessed_ui_suites_path = tidy_src.join("ui_suites.txt");
```

Comment:
This should not be the same name, otherwise we'll still bork things.

```suggestion
        let blessed_ui_suites_path = tidy_src.join("ui_suites_blessed.txt");
```